zds/utils/views.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect, get_object_or_404
+from django.utils.decorators import method_decorator
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from zds.member.decorator import can_write_and_read_now
+from zds.utils.models import CommentVote
+
+
+class KarmaView(APIView):
+    message_class = None
```

Comment:
En gros soit tu bosses sur des objets `Post`, soit tu bosses sur
`ContentReaction`(les deux héritent de la classe `Comment`), donc tu dois
récupérer l'objet pour être certain qu'il existe et avoir les infos sur
l'auteur

On Sun, Mar 27, 2016, 6:17 PM Gérard Paligot <notifications@github.com>
wrote:

> In zds/utils/views.py
> <https://github.com/zestedesavoir/zds-site/pull/3265#discussion_r57530947>
> :
>
> > @@ -0,0 +1,78 @@
> > +# coding: utf-8
> > +
> > +from django.conf import settings
> > +from django.contrib.auth.decorators import login_required
> > +from django.shortcuts import redirect, get_object_or_404
> > +from django.utils.decorators import method_decorator
> > +from rest_framework.views import APIView
> > +from rest_framework.response import Response
> > +
> > +from zds.member.decorator import can_write_and_read_now
> > +from zds.utils.models import CommentVote
> > +
> > +
> > +class KarmaView(APIView):
> > +    message_class = None
>
> Tu peux m'en dire plus ? Je vois qu'un seul usage : self.object =
> get_object_or_404(self.message_class, pk=self.kwargs.get('pk')) et je
> vois donc pas trop le rapport.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zestedesavoir/zds-site/pull/3265/files/6902a1ad548de63f39b86649c90c76e0332d95ec#r57530947>
>